GitCommands/UserRepositoryHistory/RepositoryHistoryManager.cs

```diff
@@ -0,0 +1,25 @@
+﻿namespace GitCommands.UserRepositoryHistory
+{
+    /// <summary>
+    /// Provides a convinient and centralised way of dealing with histories of local and remote repositories.
+    /// </summary>
+    public static class RepositoryHistoryManager
```

Comment:
❗️ Introduction of static state is unnecessary and should be avoided. In a separate pull request, these should be moved to MEF exports and this holder type eliminated. A bug needs to be filed so this work is not lost, and a comment added to #4768 linking to that bug.

Reply:
I am unfamiliar with MEF, and I am quite keen to learn how it can help in this case.

This implementation is very atypical of me - I normally create instance-based types+interface (for behavioural types), which can be registered with DI and all that comes with this.
Since the current codebase is not particularly ready for DI, I've been creating instances of behavioural types manually early on in the lifecycle of respective components depending on those types.

In this instance I've retained the top level static class for the convenience sake. This type is a pure infrastructural plumbing, it doesn't have any behaviours attached to it - those are provided by the respective repository managers (which are tested).

I totally understand that static components increase coupling, reduce cohesion and make things much harder to test. In this case, `RepositoryHistoryManager` is used exclusively by the UI layer, so the issue of coupling IMO is not as severe as it would be for an general purpose injectable type.
I don't have a great deal of experience in UI testing, and I am keen to learn more.

Reply:
Looks like I have a crazy two weeks coming up, but I can combine some information about MEF usage with the work to adopt vs-mef right after. I would encourage y'all to semi-avoid changes in this space because the patterns I'm planning to show you are relatively new, or at least not yet widely adopted. It's essentially a combination of the knowledge in #4768 and the caching from vs-mef to create a test environment that is accurate, reliable, and easy to use (pit of success) with little to no observable performance hit.